src/adapter/src/coord/command_handler.rs

```diff
@@ -755,4 +771,75 @@ impl Coordinator {
         let update = self.catalog().state().pack_session_update(session, -1);
         self.send_builtin_table_updates(vec![update]).await;
     }
+
+    fn handle_append_webhook(
+        &mut self,
+        database: String,
+        schema: String,
+        name: String,
+        tx: oneshot::Sender<Result<Option<AppendWebhookResponse>, AdapterError>>,
+    ) {
+        // Make sure the feature is enabled before doing anything else.
+        if !self.catalog().system_config().enable_webhook_sources() {
+            // We don't care if the listener went away.
+            let _ = tx.send(Err(AdapterError::Unsupported("enable_webhook_sources")));
+            return;
+        }
+
+        /// Attempts to resolve a Webhook source from a provided `database.schema.name` path.
+        ///
+        /// Returns a struct that can be used to append data to the underlying storate collection, and the
+        /// types we should cast the request to.
+        fn resolve(
+            coord: &Coordinator,
+            database: String,
+            schema: String,
+            name: String,
+        ) -> Option<(CollectionAppender, ColumnType, Option<ColumnType>)> {
+            // Resolve our collection.
+            let database = coord.catalog().resolve_database(&database).ok()?;
+            let schema_id = database.schemas_by_name.get(&schema)?;
+            let schema = database
+                .schemas_by_id
+                .get(schema_id)
+                .expect("found this schema by name");
+            let item_id = schema.items.get(&name)?;
+
+            // Make sure it's a webhook source.
+            let entry = coord.catalog().get_entry(item_id);
+            let (body_ty, header_ty) = match entry.item() {
+                CatalogItem::Source(source) if source.is_webhook() => {
+                    // All Webhook sources should have at most 2 columns.
+                    mz_ore::soft_assert!(source.desc.arity() <= 2);
+
+                    let body = source
+                        .desc
+                        .get_by_name(&"body".into())
+                        .map(|(_idx, ty)| ty.clone())?;
+                    let header = source
+                        .desc
+                        .get_by_name(&"headers".into())
+                        .map(|(_idx, ty)| ty.clone());
+
+                    (body, header)
+                }
+                // TODO(parkmycar): Return a specific error that indicates an object was found, but
+                // it was not a webhook source. This would probably map to an HTTP 400 BAD_REQUEST.
```

Comment:
I think a generic 400 error is preferable; including any more detail than "you can't do that here" reveals more information about the catalog than is necessary.